### ISSUE TITLE
Firmware channel update

### ISSUE THREAD

**Issue by fliescher** [fhorinek/BB#165]
I was on D232. When I want to "update" to R232 (fw channel release chosen), the system information still says D232 (fw channel selection stays at release however). Same happens when updating from D234 to R232 and also when updating using the strato.fw file from the fw repo

**Comment by vojtechBohman** [comment]
- D232 downloaded.
- Then I select Release branche.
- Clicked update.
- R was downloaded, but after update, it still says D version.
- Select manual instal of R version
- Bus fault
![image](https://user-images.githubusercontent.com/82414966/151236437-85deb641-88ee-4f1c-b430-cd19ef66f596.png)

Do you test the software or you just release and wait? Every new feature is full of bugs. Is quite boring to reset device so many times.

**Comment by thrull** [comment]
@fliescher @vojtechBohman D232 is the latest version for channel release & testing (28.01.2022 21:52 :-). 

Latest devel is D235. I fixed the "detection" and misleading download message for the update channels when the "next" version is empty (waiting to be released). 

@vojtechBohman I could not replicate the the error with manual firmware install. Are You able to replicate it now? Do You have any files on Your Strato in ./system/fw/ directory?

**Comment by vojtechBohman** [comment]
I have Release D232
- check for update
![image](https://user-images.githubusercontent.com/82414966/151849250-04dab96b-fc23-459c-a73e-1e748b0b556a.png)

- new FW detected. Download? yes
![image](https://user-images.githubusercontent.com/82414966/151849282-93a75e0a-ec78-452d-91a0-fdf8f0eac12a.png)

- immediately downloaded
- Start update? yes
![image](https://user-images.githubusercontent.com/82414966/151849335-c39cd88a-83b1-4848-b7a2-86b016a9ed46.png)

-> nothing happens
# Debug
[debug.log](https://github.com/fhorinek/BB/files/7973130/debug.log)

# FW dir
![image](https://user-images.githubusercontent.com/82414966/151849527-0d2e946e-58ae-49f7-840f-a477f8205122.png)

**Comment by thrull** [comment]
@vojtechBohman thanks for the report, this "immediate download" and related problem is already fixed in #170 & #173 - fixed will be released in next devel firmware and after tests in normal release channel.

I will try to replicate manual fw install crash locally and provide fix if possible.

**Comment by fliescher** [comment]
Hi @thrull 
I checked for updates on the Strato in Release channel but it said it's the latest fw (in both, D236 and D237 - although R238 was already published in strato.skybean.eu/fw)
When switching to the devel channel D238 is found

**Comment by thrull** [comment]
@fliescher nope, 238 is devel release (D0000238), there is no R238. Updates work as inteded for now.

FW repo contains all published versions. Update channels points to a specific version from selected "line". 

I will ask fero if we need to reorganize fw repo to be exactly in line with the update channels.

**Comment by fliescher** [comment]
thanks for the explanation! I thought strato.skybean.eu/fw only contains the release versions :)
